examples/improved_wgan.py

```diff
@@ -71,10 +71,23 @@ def gradient_penalty_loss(y_true, y_pred, averaged_samples, gradient_penalty_wei
     Note that this loss function requires the original averaged samples as input, but Keras only supports passing
     y_true and y_pred to loss functions. To get around this, we make a partial() of the function with the
     averaged_samples argument, and use that for model training."""
-    gradients = K.gradients(K.sum(y_pred), averaged_samples)
-    gradient_l2_norm = K.sqrt(K.sum(K.square(gradients)))
+    # first get the gradients:
+    #   assuming: - that y_pred has dimensions (batch_size, 1)
+    #             - averaged_samples has dimensions (batch_size, nbr_features)
+    # gradients afterwards has dimension (batch_size, nbr_features), basically
+    # a list of nbr_features-dimensional gradient vectors
+    gradients = K.gradients(y_pred, averaged_samples)[0]

```

Comment:
can you explain why only index 0 is accessed?

Reply:
gradients(y, [x1, x2, ...]) always returns a list, each entry is the gradient of y with respect to xi.
As we only pass one tensor for x (averaged_samples) and gradients can handle non-wrapped single elements the output is still in list form, but only containing this one elment. As we don't want the list but the actual element, we need the indexing.